src/main/java/org/springframework/shell/core/SimpleParser.java

```diff
@@ -425,6 +425,10 @@ private void printHintMessage(final Set<CliOption> cliOptions, Map<String, Strin
 	 * @return a non-<code>null</code> string
 	 */
 	String normalise(final String rawInput) {
+		// If input might have a quoted argument, let tokenizer handle spaces
+		if (rawInput.contains("'") || rawInput.contains("\"")) {
+			return rawInput.trim();
```

Comment:
This function was only used for sanitizing whitespace, and the tokenizer is capable of handling extra spaces between arguments. I'll add an extra test case with both extraneous spaces and a quoted argument like `  bar    --option1       "boom"  ` to verify that it parses it correctly.